cypress/support/commands.js

```diff
@@ -24,16 +24,181 @@
 //
 // -- This will overwrite an existing command --
 // Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
-Cypress.Commands.add('verifyCode',(code, verificationUrl)=>{
-    cy.visit(`https://trakt.tv/auth/signin`);
-    cy.get('#user_login').type(Cypress.env('user').username);
-    cy.get('#user_password').type(Cypress.env('user').password);
-    cy.contains('Sign in').click();
-
-    cy.visit(verificationUrl);
-	cy.get('#code').type(code);
-	cy.contains('Continue').click();
-    cy.get('#auth-form-wrapper').should('be.visible')
-	cy.get('input[name="commit"]').contains('Yes').click();
-	cy.contains('Woohoo!').should('be.visible');
-})
\ No newline at end of file
+Cypress.Commands.add('verifyCode', (code, verificationUrl) => {
+	cy.visit(`https://trakt.tv/auth/signin`)
+	cy.get('#user_login').type(Cypress.env('user').username)
+	cy.get('#user_password').type(Cypress.env('user').password)
+	cy.contains('Sign in').click()
+
+	cy.visit(verificationUrl)
+	cy.get('#code').type(code)
+	cy.contains('Continue').click()
+	cy.get('#auth-form-wrapper').should('be.visible')
+	cy.get('input[name="commit"]').contains('Yes').click()
+	cy.contains('Woohoo!').should('be.visible')
+})
+
+Cypress.Commands.add('getAccessToken', () => {
+	cy.request({
+		method: 'POST',
+		url: 'https://api.trakt.tv/oauth/device/code',
+		headers: {
+			'Content-Type': 'application/json',
+		},
+		failOnStatusCode: false,
+		body: {
+			client_id: Cypress.env('client_id'),
+		},
+	}).then($response => {
+		expect($response.status).to.be.eq(200)
+		expect($response.body).to.have.property('user_code')
+		expect($response.body).to.have.property('device_code')
+		expect($response.body).to.have.property('verification_url')
+		expect($response.body).to.have.property('expires_in')
+		expect($response.body).to.have.property('interval')
+
+		cy.verifyCode(
+			$response.body.user_code,
+			$response.body.verification_url
+		).then(() => {
+			cy.request({
+				method: 'POST',
+				url: 'https://api.trakt.tv/oauth/device/token',
+				headers: {
+					'Content-Type': 'application/json',
+				},
+				body: {
+					code: $response.body.device_code,
+					client_id: Cypress.env('client_id'),
+					client_secret: Cypress.env('client_secret'),
+				},
+			}).then($tokenResponse => {
+				expect($tokenResponse.status).to.be.eq(200)
```

Comment:
Remove assertions from here, as there is a specific test where this is done. Here in custom commands, we just "believe" that this is going to work and return the response at the end where we can make an assertion in the test.